propagate_const.h

```diff
@@ -73,7 +80,24 @@ class propagate_const {
 
   PROPAGATE_CONST_CONSTEXPR propagate_const(propagate_const&& p) = default;
 
-#ifdef PROPAGATE_CONST_HAS_EXPRESSION_SFINAE
+#ifdef PROPAGATE_CONST_HAS_NO_EXPRESSION_SFINAE
+  //
+  // Make converting constructors explicit as we cannot use SFINAE to check.
+  //
+  template <class U, class = enable_if_t<is_constructible<T, U&&>::value>>
```

Comment:
Have you tried with 

```
template <class U, enable_if_t<is_constructible<T, U&&>::value, int>=0>
```
?

Reply:
Or is this exactly what MSVC donesn't like?

Reply:
I’m happy to iterate and improve.